uw-research-computing/configure-ssh.md

```diff
@@ -0,0 +1,111 @@
+---
+highlighter: none
+layout: markdown-page
+title: Automating CHTC Log In
+---
+
+This guide describes 
+* how to authenticate with Duo when logging into CHTC
+* how to set your login (ssh) configuration to “reuse” a two factor authenticated 
+connection over a certain period of time. 
+
+## Authentication with Duo
+
+As of December 2022, accessing CHTC resources 
+now requires two factor authentication. The first “factor” uses your NetID password 
+(or ssh keys) and the second “factor” is authentication using Duo, via either a 
+Duo fob or the Duo app. 
+
+See the following video for an demonstration of two factor authentication with Duo 
+when logging into CHTC: 
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/J-wxsrQ3v04" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Re-Using SSH Connections
+
+To reduce the number of times it is necessary to enter your credentials, it’s possible 
+to customize your ssh configuration in a way that allows you to “reuse” a connection 
+for logging in again or moving files.  This configuration is totally optional, and 
+most useful if you will connect to 
+the same server multiple times in a short window, for example, when uploading or 
+downloading files. 
+
+**WARNING:** This guide describes how to configure your local machine to not require 
+reentering your NetID password or Duo authentication each time you login. 
+This should **ONLY** be used on secure devices that you manage - **it should 
+not be used on any shared laptop, desktop, or research group resource. Users 
+found violating this policy risk having their CHTC account permanently deactivated.** 
+
+The instructions below are meant for users who can use a terminal (Mac, Linux): 
+
+1. Open a terminal window.
+
+1. Create (or edit) your personal ssh configuration file at `~/.ssh/config` to use 
+what’s called "[ControlMaster](https://www.anchor.com.au/blog/2010/02/ssh-controlmaster-the-good-the-bad-the-ugly/)" (third party blog link). 
+Executing the code below will add the right information to the `config` file; you 
+can also copy and paste the content in quotes into your config file directly. 
+	```	
+	# Let's create (or add to) our SSH client
+	# configuration file. 
+	echo "
+	Host *.chtc.wisc.edu
+	  # Turn ControlMaster on
+	  ControlMaster auto
+	  # ControlMaster connection will persist
+	  # for 2 hours of idleness, after which
+	  # it will disconnect
+	  ControlPersist 2h
+	  # Where to store files that represent
+	  # the ControlMaster persistent connections
+	  ControlPath ~/.ssh/connections/%r@%h:%p" >> ~/.ssh/config
+	```
+	{:.term}
+
+1. You also create a directory that will be used to track connections. In 
+the same .ssh directory, make a folder called `connections` by typing: 
+```
+$ mkdir -p ~/.ssh/connections
+```{: .term}
```

Comment:
Should there be an empy space between line 68/69? Unsure if this is meant to be in-text code or a code chunk (either way is fine)